core/indexed/lazy_slice.go

```diff
@@ -0,0 +1,46 @@
+package indexed
+
+import (
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/encoder"
+)
+
+type LazySlice[T any] struct {
+	indexes []int
+	data    []byte
+}
```

Comment:
A small description will help other understand this data structure easily

Reply:
Added description with an example